src/main/kotlin/com/openlattice/organizations/DatasetController.kt

```diff
@@ -167,11 +171,20 @@ class DatasetController : DatasetApi, AuthorizingComponent {
             @PathVariable(ID) organizationId: UUID,
             @RequestBody tableNames: Set<String>) {
         val tableIdByFqn = getExternalDatabaseObjectIdByFqnMap(organizationId, tableNames)
-        tableIdByFqn.forEach { ensureObjectCanBeDeleted(it.value) }
-        val aclKeys = tableIdByFqn.map { AclKey(it.value) }
+        val tableIds = tableIdByFqn.values
+        val aclKeys = tableIds.map { AclKey(it) }.toSet()
         aclKeys.forEach { aclKey ->
             ensureOwnerAccess(aclKey)
         }
+        tableIds.forEach { tableId ->
+            ensureObjectCanBeDeleted(tableId)
+            val columnIds = edms.getExternalDatabaseTableWithColumns(tableId).columns.map { it.id }.toSet()
+            columnIds.forEach { ensureObjectCanBeDeleted(it) }
+            val aclKeys = columnIds.map { AclKey(tableId, it) }.toSet()
+            aclKeys.forEach { aclKey ->
```

Comment:
This can be also bulked the same way as in `getAuthorizedTableIds`, but rather then returning the authorized ones, you can compare if their size matches or not and throw a `ForbiddenException` if not

Reply:
I'll still need `ensureObjectCanBeDeleted` though right?

Reply:
That's right